src/store.ts

```diff
@@ -0,0 +1,113 @@
+import { types } from "mobx-state-tree";
+
+const validateMail = (str: string): boolean => {
+  let atIndex = str.indexOf("@");
+  let dotIndex = str.lastIndexOf(".");
+  return atIndex > 0 && dotIndex > atIndex;
+};
+
+const User = types.model({
+  name: types.string,
+  mail: types.string,
+  complaint: types.string,
+  selectedType: types.maybeNull(
+    types.enumeration(["Service related", "Product related"])
+  ),
+  createdAt: types.maybe(types.Date),
+});
+
+export const RootStore = types
+  .model({
+    name: types.maybe(types.string),
+    mail: types.maybe(types.string),
+    complaint: types.maybe(types.string),
+    selectedType: types.maybeNull(
+      types.enumeration(["Service related", "Product related"])
```

Comment:
These values should not be hard code, it has to come from i18n

Reply:
It will be addressed in the next PR.